src/v0/destinations/impact_radius/transform.js

```diff
@@ -0,0 +1,285 @@
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Basic ${btoa(`${Config.accountSID}:${Config.apiKey}`)}`
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.FORM = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  throw new TransformationError(
+    "Payload could not be populated due to wrong input"
+  );
+};
+
+const buildPageLoadResponse = (
+  message,
+  campaignId,
+  impactAppId,
+  enableEmailHashing
+) => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.PAGELOAD.name]
+  );
+  if (isDefinedAndNotNull(payload.CustomerEmail)) {
+    payload.CustomerEmail = enableEmailHashing
+      ? sha1(payload?.CustomerEmail)
+      : payload?.CustomerEmail;
+  }
+  payload.CampaignId = campaignId;
+  if (isDefinedAndNotNullAndNotEmpty(impactAppId)) {
+    payload.PropertyId = impactAppId;
+    payload.ImpactAppId = impactAppId;
+  }
+
+  const os = get(message, "context.os.name");
+  if (os && isAppleFamily(os.toLowerCase())) {
+    payload.AppleIfv = get(message, "context.device.id");
+    payload.AppleIfa = get(message, "context.device.advertisingId");
+  } else if (os && os.toLowerCase() === "android") {
+    payload.AndroidId = get(message, "context.device.id");
+    payload.GoogAId = get(message, "context.device.advertisingId");
+  }
+  return payload;
+};
+
+/**
+ * This function is used to build the response for identify call.
+ * @param {*} message
+ * @param {*} category
+ * @param {*} Config
+ * @returns
+ */
+const identifyResponseBuilder = (message, Config) => {
+  const {
+    campaignId,
+    enableIdentifyEvents,
+    impactAppId,
+    enableEmailHashing
+  } = Config;
+
+  if (!enableIdentifyEvents) {
+    throw new ConfigurationError(
+      `${message.type} events are disabled from Config`
+    );
+  }
+
+  return responseBuilder(
+    buildPageLoadResponse(message, campaignId, impactAppId, enableEmailHashing),
+    CONFIG_CATEGORIES.PAGELOAD.endPoint,
+    Config
+  );
+};
+
+/**
+ * This function is used to build the response for track call.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const trackResponseBuilder = (message, Config) => {
+  const {
+    campaignId,
+    eventTypeId,
+    impactAppId,
+    enableEmailHashing,
+    actionEventNames,
+    installEventNames,
+    rudderToImpactProperty,
+    productsMapping
+  } = Config;
+
+  let eventType;
+
+  if (installEventNames && Array.isArray(installEventNames)) {
+    installEventNames.forEach(event => {
+      if (event.eventName === message.event) {
+        eventType = "install";
+      }
+    });
+  }
+  if (actionEventNames && Array.isArray(actionEventNames)) {
+    actionEventNames.forEach(event => {
+      if (event.eventName === message.event) {
+        eventType = "action";
+      }
+    });
+  }
```

Comment:
```suggestion
  if (
    installEventNames &&
    Array.isArray(installEventNames) &&
    installEventNames.includes({ eventName: message.event })
  ) {
    eventType = "install";
  }
  if (
    actionEventNames &&
    Array.isArray(actionEventNames) &&
    actionEventNames.includes({ eventName: message.event })
  ) {
    eventType = "action";
  }
```

Reply:
If an event is present in both `installEventNames` and `actionEventNames`, then you're only sending one event with `eventType = "action"`. Is this fine?

Reply:
Yes

Reply:
If `message.event` is present in both `installEventNames` and `actionEventNames`, you're sending only one event with `eventType = "action"`. Shouldn't you send two events, one with `eventType` equals to `install` and other with `action`?
cc @ItsSudip

Reply:
I tried to align with what segment is doing here.

Reply:
We should send two events. As they are mentioning event names in two places, they should want to send both the install and action events.